### PR TITLE
249 fix hard coded point scale factor

### LINKED ISSUE #249: Bug: Scale-Factor hardcoded to 2, causing position of points to not change with scaling of graph
Scale-Factor hardcoded to 2, causing the position of points to not change with scaling of graph
image

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #249

# What was the issue
The scale value for scaling the graph was hard coded to the point, leading to the point staying in the same position even when the graph is scaling.

# What was done and why
Allow scale to be passed in when creating data point positions so that the position of points will change relative to the graph scaling.

# How it was tested
- Update existing unit test

[Screencast from 2024-03-31 02:23:48 PM.webm](https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/assets/94843824/204e6280-1f2c-4c3e-8747-679f49af6837)

### CODE REVIEW

**Review by Subzeero — COMMENTED** [review]
Nice, that video looks good 🙌

There's a few bits of commented out code that we should probably remove before merge?

**Comment by LongQuanNguyen** [comment]
I notice GenerateXYZ also takes in hardcoded maxValues. I have modified it to get them form DB now. As a result, when u first run the app, there will be no tick on the axis until the columns are selected.

https://github.com/UniversityOfSaskatchewanCMPT371/term-project-2024-team-2/assets/94843824/f2ae5d43-baf1-4804-9dc8-0102b8f1d0a1

File: my-webxr-app/tests/components/GenerateXYZ.test.tsx [hunk 1]
```diff
@@ -1,77 +1,106 @@
...
+  //   expect(yAxisTicks).toHaveLength(21);
+  //   expect(yAxisTicks.map((e) => e.children[1].props.text))
+  //     .toEqual([-10, -9, -8, -7, -6, -5, -4, -3, -2, -1,
+  //       0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map(String));
+  //
+  //   const zAxisTicks = axes[2].children.filter((e) => e.instance.name === 'tick');
+  //   expect(zAxisTicks).toHaveLength(21);
+  //   expect(zAxisTicks.map((e) => e.children[1].props.text))
+  //     .toEqual([-100, -90, -80, -70, -60, -50, -40, -30, -20, -10,
+  //       0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100].map(String));
```
Comment (LongQuanNguyen) [inline my-webxr-app/tests/components/GenerateXYZ.test.tsx:104]: Waiting to pair with Zander on why the max values are not returned here, leading to no ticks getting rendered.

File: my-webxr-app/src/App.tsx [hunk 2]
```diff
@@ -16,23 +15,37 @@ import { rollbarConfig } from './utils/LoggingUtils';
...
+//   [2, 2, 2],
+//   [3, 3, 3],
+//   [4, 4, 4],
+//   [5, 5, 5],
+//   [6, 6, 6],
+//   [7, 7, 7],
+//   [8, 8, 8],
+//   [9, 9, 9],
+//   [10, 10, 10],
+// ];
```
Comment (matthew-buglass) [inline my-webxr-app/src/App.tsx:34]: Remove the commented-out code
```suggestion
```

File: my-webxr-app/src/App.tsx [hunk 3]
```diff
@@ -16,23 +15,37 @@ import { rollbarConfig } from './utils/LoggingUtils';
...
-// endPoint is used to determine what axis is being calculated, should not need to change
-// adjust the size of the tube, shouldn't need to change unless
 const radius: number = 0.002;
 
 Dexie.delete('CsvDataBase');
 const DAL = getDatabase() as DataAbstractor;
+// await DAL.storeCSV(batch);
+// await DAL.calculateStatistics();
+// await DAL.storeStandardizedData();
+// await DAL.storePCA(await DAL.getAvailableFields());
```
Comment (matthew-buglass) [inline my-webxr-app/src/App.tsx:48]: Remove the commented-out code
```suggestion
```

File: my-webxr-app/tests/components/GenerateXYZ.test.tsx [hunk 4]
```diff
@@ -1,77 +1,106 @@
-import ReactThreeTestRenderer, { waitFor } from '@react-three/test-renderer';
-import { Vector3 } from 'three';
+import ReactThreeTestRenderer from '@react-three/test-renderer';
+// import { Vector3 } from 'three';
```
Comment (matthew-buglass) [inline my-webxr-app/tests/components/GenerateXYZ.test.tsx:2]: Remove the commented-out code
```suggestion
```

File: my-webxr-app/tests/components/GenerateXYZ.test.tsx [hunk 5]
```diff
@@ -1,77 +1,106 @@
...
 import { XR } from '@react-three/xr';
-import { ReactThreeTestInstance } from '@react-three/test-renderer/dist/declarations/src/createTestInstance';
+// eslint-disable-next-line max-len
+// import { ReactThreeTestInstance } from '@react-three/test-renderer/dist/declarations/src/createTestInstance';
```
Comment (matthew-buglass) [inline my-webxr-app/tests/components/GenerateXYZ.test.tsx:5]: Remove the commented-out code
```suggestion
```

**Review by matthew-buglass — CHANGES_REQUESTED** [review]

**Review by Subzeero — APPROVED** [review]
Looks good 🙌

**Pull request merged**